### ISSUE TITLE
blt_add_benchmark should take CONFIGURATIONS and OMP_NUM_THREADS arguments

### ISSUE THREAD

**Issue by arghdos** [LLNL/blt#635]
E.g., here:

https://github.com/LLNL/blt/blob/553a90261dbe0e2c064330d9a129fa084267a69c/cmake/BLTMacros.cmake#L937

Not only is this consistent with `blt_add_test`, but this enables e.g., creating your own configurations that specific benchmarks should be run on.  For instance, I would like to create a "Multinode" configurations where only tests / benchmarks that require multiple nodes should be run (e.g., to benchmark inter-node bandwidth).  This way I can create a Jenkins job or the like that _only_ runs those tests when I allocate two+ nodes for a job.

Today, I have to do something like:

```
blt_add_test(...
                    CONFIGURATIONS Benchmark Multinode
)
```

I think this would be a minimally invasive PR, and I'd be happy to work on it if y'all agree.

**Comment by arghdos** [comment]
Looking at it, it appears that NUM_OMP_THREADS is missing as well

**Comment by white238** [comment]
That's a great idea. We appreciate any help if you have time, otherwise I will handle it shortly.

**Comment by arghdos** [comment]
started in #636

**Comment by white238** [comment]
Closed in #636
